### PR TITLE
build: missing newline when printing build details on error

### PR DESCRIPTION
**Pull request by crazy-max**
Before

```
foo@bar:~ $ docker buildx build .
...
------
Dockerfile:356
--------------------
 355 |     ARG TARGETPLATFORM
 356 | >>> RUN --mount=type=cache,sharing=locked,id=moby-rootlesskit-aptlib,target=/var/lib/apt \
 357 | >>>     --mount=type=cache,sharing=locked,id=moby-rootlesskit-aptcache,target=/var/cache/apt \
 358 | >>>         apt-get update && xx-apt-get install -y --no-install-recommends \
 359 | >>>             gcc libc6-dev
 360 |     ENV GO111MODULE=on
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update && xx-apt-get install -y --no-install-recommends             gcc libc6-dev" did not complete successfully: exit code: 100

View build details: docker-desktop://dashboard/build/default/default/oj5lusuq2n0ufovvphbaxfjn0foo@bar:~ $
```

After

```
foo@bar:~ $ docker buildx build .
...
------
Dockerfile:356
--------------------
 355 |     ARG TARGETPLATFORM
 356 | >>> RUN --mount=type=cache,sharing=locked,id=moby-rootlesskit-aptlib,target=/var/lib/apt \
 357 | >>>     --mount=type=cache,sharing=locked,id=moby-rootlesskit-aptcache,target=/var/cache/apt \
 358 | >>>         apt-get update && xx-apt-get install -y --no-install-recommends \
 359 | >>>             gcc libc6-dev
 360 |     ENV GO111MODULE=on
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update && xx-apt-get install -y --no-install-recommends             gcc libc6-dev" did not complete successfully: exit code: 100

View build details: docker-desktop://dashboard/build/default/default/oj5lusuq2n0ufovvphbaxfjn0
foo@bar:~ $
```

### CODE REVIEW

File: tests/build.go [hunk 1]
```diff
@@ -187,6 +190,42 @@ RUN busybox | head -1 | grep v1.35.0
 	require.NoError(t, cmd.Run())
 }
 
+func testBuildDetailsLink(t *testing.T, sb integration.Sandbox) {
+	buildDetailsPattern := regexp.MustCompile(`(?m)^View build details: docker-desktop://dashboard/build/[^/]+/[^/]+/[^/]+\n$`)
+
+	// create desktop-build .lastaccess file
+	home, err := os.UserHomeDir() // TODO: sandbox should create a temp home dir and expose it through its interface
```
Comment (jedevc) [inline tests/build.go:197]: I think this should still be "safe" to merge without this TODO?

No other tests access this file, so should be alright for now? Though obviously good to follow up asap.

File: tests/build.go [hunk 2]
```diff
@@ -187,6 +190,42 @@ RUN busybox | head -1 | grep v1.35.0
...
+		if err := os.Remove(dblaFile.Name()); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// build simple dockerfile
+	dockerfile := []byte(`FROM busybox:latest
+RUN echo foo > /bar`)
+	dir := tmpdir(t, fstest.CreateFile("Dockerfile", dockerfile, 0600))
+	cmd := buildxCmd(sb, []cmdOpt{withArgs("build", "--output=type=cacheonly", dir)}...)
```
Comment (jedevc) [inline tests/build.go:214]: nit:
```suggestion
	cmd := buildxCmd(sb, withArgs("build", "--output=type=cacheonly", dir))
```

File: tests/build.go [hunk 3]
```diff
@@ -187,6 +190,42 @@ RUN busybox | head -1 | grep v1.35.0
...
 
+func testBuildDetailsLink(t *testing.T, sb integration.Sandbox) {
+	buildDetailsPattern := regexp.MustCompile(`(?m)^View build details: docker-desktop://dashboard/build/[^/]+/[^/]+/[^/]+\n$`)
+
+	// create desktop-build .lastaccess file
+	home, err := os.UserHomeDir() // TODO: sandbox should create a temp home dir and expose it through its interface
+	require.NoError(t, err)
+	dbDir := path.Join(home, ".docker", "desktop-build")
+	require.NoError(t, os.MkdirAll(dbDir, 0755))
+	dblaFile, err := os.Create(path.Join(dbDir, ".lastaccess"))
```
Comment (jedevc) [inline tests/build.go:201]: Could we run the simple/erroneous builds before this as well, and check that we *don't* get the build details? Then we have a test to make sure this doesn't leak into non desktop-build scenarios.

File: tests/build.go (hunk 1 above)
Comment (crazy-max) [reply tests/build.go:197]: Yes fine for now 👍

File: tests/build.go (hunk 3 above)
Comment (crazy-max) [reply tests/build.go:201]: Good point will add this

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
No need to release-note this, since the original feature wasn't under the release notes.

**Pull request merged**